# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Minahil973/MinahilAmir-week1/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

Lane: Refresh / Content Opportunity Scoring

I chose the Refresh / Content Opportunity Scoring lane because it focuses on identifying pages that should be reviewed first based on real search and content performance signals. Instead of simply training a machine learning model, this project supports decision-making by helping content teams prioritize which pages need attention. I am interested in understanding how data can be used to rank opportunities and provide practical recommendations supported by evidence.

In [6]:
!git clone https://github.com/Minahil973/MinahilAmir-week1.git

Cloning into 'MinahilAmir-week1'...
remote: Enumerating objects: 124, done.
remote: Counting objects: 100% (124/124), done.
remote: Compressing objects: 100% (78/78), done.
remote: Total 124 (delta 40), reused 99 (delta 30), pack-reused 0 (from 0)
Receiving objects: 100% (124/124), 1.84 MiB | 4.89 MiB/s, done.
Resolving deltas: 100% (40/40), done.


In [9]:
!pwd
!ls
!find . -name "*.csv"

/content
MinahilAmir-week1  sample_data
./MinahilAmir-week1/outputs/refresh_queue_sample.csv
./MinahilAmir-week1/data/raw/content_refresh_anonymized.csv
./sample_data/california_housing_train.csv
./sample_data/california_housing_test.csv
./sample_data/mnist_train_small.csv
./sample_data/mnist_test.csv


In [10]:
import pandas as pd

df = pd.read_csv("/content/MinahilAmir-week1/data/raw/content_refresh_anonymized.csv")

print("Rows:", len(df))
print("Columns:", len(df.columns))

df.head()

Rows: 30000
Columns: 44


,content_id,client_id,search_volume,competition,competition_level,cpc,content_type,main_intent,word_count,char_count,...,char_count_tier,ctr,avg_position,engagement_rate,scroll_rate,ai_traffic_pct,impression_tier,position_tier,trend_direction,trend_pct
0,content_304f48230142,client_f369cb89fc,10.0,0.67,HIGH,2.05,keyword article,transactional,3221.0,20457.0,...,15000-25000,0.76,10.6,5.88,4.55,0.0,good,striking,down,-41.4
1,content_a1fb4e703a9e,client_4e07408562,90.0,0.01,LOW,0.05,keyword article,informational,2481.0,15562.0,...,15000-25000,0.05,20.3,0.00,10.00,0.0,good,page_3_5,down,-57.7
2,content_9aa793d4d895,client_7f2253d7e2,0.0,0.00,LOW,0.00,keyword article,informational,3515.0,23643.0,...,15000-25000,0.09,36.5,0.00,28.57,0.0,good,page_3_5,down,-60.9
3,content_331d6c4de07b,client_19581e27de,10.0,0.00,LOW,0.00,keyword article,commercial,NaN,NaN,...,NaN,0.49,6.2,1.28,3.45,0.0,good,page_1,stable,-13.8
4,content_d99b7a2d90ca,client_3fdba35f04,0.0,0.00,LOW,0.00,keyword article,informational,2803.0,17469.0,...,15000-25000,0.13,44.0,0.00,24.29,0.0,good,page_3_5,down,-34.7


In [11]:
print("Dataset Shape:", df.shape)
print("\nColumns:\n")
print(df.columns)

Dataset Shape: (30000, 44)

Columns:

Index(['content_id', 'client_id', 'search_volume', 'competition',
       'competition_level', 'cpc', 'content_type', 'main_intent', 'word_count',
       'char_count', 'provider_used', 'model_used', 'impressions_90d',
       'clicks_90d', 'pageviews_90d', 'sessions_90d', 'users_90d',
       'engaged_sessions_90d', 'ai_sessions_90d', 'scroll_events_90d',
       'days_with_impressions', 'days_with_sessions', 'impressions_last_30d',
       'clicks_last_30d', 'sessions_last_30d', 'impressions_prev_30d',
       'clicks_prev_30d', 'sessions_prev_30d', 'content_age_days', 'age_tier',
       'age_tier_order', 'days_since_last_update', 'freshness_tier',
       'word_count_tier', 'char_count_tier', 'ctr', 'avg_position',
       'engagement_rate', 'scroll_rate', 'ai_traffic_pct', 'impression_tier',
       'position_tier', 'trend_direction', 'trend_pct'],
      dtype='object')


In [12]:
print("Total Pages:", len(df))
print("\nTrend Direction:")
print(df["trend_direction"].value_counts())

print("\nAverage CTR:")
print(round(df["ctr"].mean(), 2))

Total Pages: 30000

Trend Direction:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64

Average CTR:
0.51


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

Decision:
The goal is to decide which content pages should be prioritized for refresh based on their search performance and content metrics.

Action:
The content or SEO team will review and update the highest-priority pages first instead of spending time reviewing every page manually.

Cost of a Wrong Call:
If a page that does not need attention is prioritized, time and resources are wasted. If an important page is missed, it may continue losing search visibility, traffic, and user engagement.

Why this decision matters:
The dataset contains performance indicators such as impressions, clicks, CTR, engagement, and content age, which can help identify pages that are more likely to benefit from a content refresh.

In [13]:
# Supporting numbers for the research question

print("Total Pages:", len(df))
print("Pages with Downward Trend:", (df["trend_direction"] == "down").sum())
print("Average CTR:", round(df["ctr"].mean(), 2))

Total Pages: 30000
Pages with Downward Trend: 16262
Average CTR: 0.51


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

The starter dataset contains 30,000 content pages with 44 features.

Three observations support choosing the Refresh / Content Opportunity Scoring lane:

- The dataset contains 30,000 pages, providing enough data to identify meaningful refresh opportunities.
- 16,262 pages have a downward trend, suggesting that many pages may need review or updating.
- The average CTR is 0.51, indicating there is room to investigate which factors are associated with better or worse performance.

These observations suggest that prioritizing pages for content refresh is a practical decision-making problem rather than simply a machine learning exercise.

In [14]:
# Quick look at the data

print("Dataset Shape:", df.shape)
print("Total Pages:", len(df))
print("Pages with Downward Trend:", (df["trend_direction"] == "down").sum())
print("Average CTR:", round(df["ctr"].mean(), 2))

Dataset Shape: (30000, 44)
Total Pages: 30000
Pages with Downward Trend: 16262
Average CTR: 0.51


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

My work can identify patterns in the data and recommend which content pages should be reviewed first. It is intended to support decision-making by highlighting pages that may benefit from a content refresh based on observed performance metrics.

However, this project cannot prove that refreshing a page will increase traffic or rankings. It also cannot predict Google's ranking algorithm or establish cause-and-effect relationships. The results should be used as decision support, not as guaranteed outcomes.

In [15]:
# Simple check
print("Dataset loaded successfully:", df.shape)

Dataset loaded successfully: (30000, 44)


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.